src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -93,8 +112,21 @@ protected Executor tryResolve() throws Exception {
                 Future<Queue.Executable> future = item.getFuture().getStartCondition();
 
                 if (!future.isDone()) {
-                    // TODO JENKINS-26130 we might be able to detect that the item is blocked on an agent which has been deleted (not just offline), and abort ourselves
-                    LOGGER.log(Level.FINER, "{0} not yet started", item);
+                    Queue.Task task = item.task;
+                    Label nodeNameLabel = task.getAssignedLabel();  // Once execution started, this was set to the Computer name
+                    // See javadocs for ExecutorStepExecution.PlaceholderTask.label
+                    long waitTime = System.currentTimeMillis()-startTimeMillis;
+                    // TODO see if we need to do something with the ExecutorStepExecution.PlaceholderTask.runningTasks
+                    if (nodeNameLabel != null && task.getLastBuiltOn() == null) {  // Can't find node it was built on!
+                        if (waitTime > TIMEOUT_WAITING_FOR_EPHMERAL_NODE && task instanceof ExecutorStepExecution.PlaceholderTask) {
+                            Queue.getInstance().cancel(item);
+                            throw new AbortException(MessageFormat.format("Killed {0} after waiting for {1} millis because we assume unknown Node {1} (ephemeral={2}) is never going to reappear!",
+                                    new Object[]{item, waitTime, nodeNameLabel.toString(), isEphemeral()}));
+                        }
+                    } else {
+                        // TODO JENKINS-26130 we might be able to detect that the item is blocked on an agent which has been deleted (not just offline), and abort ourselves
```

Comment:
This comment can be deleted now, as it was referring to the exact condition you purport to fix in this PR!